domain/src/test/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointControllerTest.kt

```diff
@@ -252,17 +295,45 @@ class ExplorationCheckpointControllerTest {
 
   @Test
   fun testController_saveCheckpoint_deleteSavedCheckpointFromDifferentProfile_isFailure() {
-    saveCheckpoint(firstTestProfile, index = 0).toLiveData()
+    saveCheckpoint(firstTestProfile, index = 0)
 
-    val deleteCheckpointLiveData =
-      deleteCheckpointWithExplorationId(secondTestProfile, index = 0).toLiveData()
+    val deleteCheckpointLiveData = explorationCheckpointController.deleteSavedExplorationCheckpoint(
+      secondTestProfile,
+      createExplorationIdForIndex(0)
+    ).toLiveData()
     deleteCheckpointLiveData.observeForever(mockResultObserver)
     verifyMockObserverIsFailure(mockResultObserver, resultCaptor)
 
     assertThat(resultCaptor.value.getErrorOrNull()).hasMessageThat()
       .contains("No saved checkpoint with explorationId")
   }
 
+  @Test
+  fun testController_saveCompatibleCheckpoint_checkCheckpointIsCompatible() {
+    explorationCheckpointTestHelper.saveCheckpointForFractionsStory0Exploration0(
+      profileId = firstTestProfile,
+      version = FRACTIONS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+    explorationCheckpointController.retrieveExplorationCheckpoint(
+      firstTestProfile,
+      FRACTIONS_EXPLORATION_ID_0
+    ).toLiveData().observeForever(mockExplorationCheckpointObserver)
+    verifyMockObserverIsSuccessful(mockExplorationCheckpointObserver, explorationCheckpointCaptor)
+  }
+
+  @Test
+  fun testController_saveInCompatibleCheckpoint_checkpointCheckpointIsNotCompatible() {
+    explorationCheckpointTestHelper.saveCheckpointForFractionsStory0Exploration0(
+      profileId = firstTestProfile,
+      version = FRACTIONS_STORY_0_EXPLORATION_0_INCORRECT_VERSION
```

Comment:
Suggest instead: ``FRACTIONS_STORY_0_EXPLORATION_0_OLD_VERSION`` and ``FRACTIONS_STORY_0_EXPLORATION_0_CURRENT_VERSION`` names to be more specific.

Reply:
done